Cargo.toml

```diff
@@ -1,4 +1,68 @@
+[package]
+name = "bob_management"
+version.workspace = true
+default-run = "backend"
+edition = "2021"
+description = "Bob Management GUI"
+authors = ["Romanov Simeon ArchArcheoss@proton.me"]
+license = "Unlicense"
+repository = "https://github.com/qoollo/bob-management"
+publish = false
+keywords = [ "BOB", "Management", "GUI" ]
+readme = "./README.md"
+
+[workspace.package]
+version = "0.0.0"   # Bob GUI version
+
+[[bin]]
+name = "backend"
+path = "backend/src/main.rs"
+required-features = [ "backend" ]
+
+[dependencies]
+# CLI
+cli = { path = "./cli", optional = true }
+
+# Backend Lib
+backend = { path = "./backend", optional = true }
+
+# Backend `main.rs` deps.
+tokio = { version = "1.32", features = ["rt", "macros", "rt-multi-thread"], optional = true }
```

Comment:
Why all these dependencies are declared here? They are `backend` and `cli` level dependencies only. On the top level Cargo.toml they are not needed

Reply:
They were needed in order to run `cargo run` in root workspace, the backend bin was defined here.... But apparently, if you define `default-run` in sub-workspace's `Cargo.toml`, that `default-run` will be inherited by root's `Cargo.toml`.
I'm not sure if this is a good thing, that sub-workspace can affect what happens at the root's workspace, but since there is no other way around, I will default to this.

Reply:
Regarding dummy binary: We could move all the package info into `backend` directory and declare root as virtual workspace (only `[worspace]` will be defined in root's `Cargo.toml`) but I personally don't like it

Reply:
Another way to solve to resolve it is to move `backend`'s src/ folder on root's level and get rid of backend folder, come to think of this...

Reply:
In fact, it looks like a virtual workspace might work here. We can think about `backend` and `frontend` as a 2 separate packages. In this case we don't need a build `[features]` at all, the packages by itself are features. We can build the backend with `cargo build --package backend` and the frontend with `cargo build --package frontend`. When we build the backend, cargo puts final binaries inside `targets` directory, but `frontend` is build with custom script so we can just reproduce the same behavior manually (copy build results to appropriate directory inside `targets`). This approach resolves all the problems: we can build `backend` and `frontend` independently, as well as build them together with normal `cargo build` command, and the output of these builds will be in the same output directory